stubs/pika/pika/amqp_object.pyi

```diff
@@ -0,0 +1,15 @@
+from typing import ClassVar
+
+class AMQPObject:
+    NAME: ClassVar[str]
+    INDEX: ClassVar[int | None]
+    def __eq__(self, other: object) -> bool: ...
```

Comment:
It will raise `AttributeError` if it's compared to an object that doesn't have a `__dict__`.

```suggestion
    def __eq__(self, other: Any) -> bool: ...
```

We could also consider this -- it looks like it's probably only meant to be compared with other `AMQPObject`s:

```suggestion
    def __eq__(self, other: AMQPObject | None) -> bool: ...  # type: ignore[override]
```